src/CommunityVoices/App/Api/Component/FileProcessor.php

```diff
@@ -0,0 +1,207 @@
+         $unpairedQuotes = [];
+         $validIdentifiers = ["allIdentifiers" => []]; // allow selection for unpaired quotes
+         $formattedSourceNames = array_map(array($this,'cleanString'),self::BATCH_SOURCE_DATA);
+         $formattedQuoteNames = array_map(array($this,'cleanString'),self::BATCH_QUOTE_DATA);
+         $sourceNameMapper = [];
+         $quoteNameMapper = [];
+         for ($i = 0; $i < count($formattedSourceNames); $i++) {
+             $sourceNameMapper[$formattedSourceNames[$i]] = self::BATCH_SOURCE_DATA[$i];
+         }
+
+         $quoteNameMapper = [];
+         for ($i = 0; $i < count($formattedQuoteNames); $i++) {
+             $quoteNameMapper[$formattedQuoteNames[$i]] = self::BATCH_QUOTE_DATA[$i];
+         }
+
+         // both of these mapper arrays are required in order to associate a formatted column with its original column to display original column name on page
+
+
+         // There will be errors/warnings on three levels: top level (column names), source level (relating to source info), quotes level (relating to quotes info)
+         // any errors on the top level will require a re upload
+
+         // NOTE: Skip the second row of the source file due to format given by John
+         // first pass through source sheet, creating entry for each interview. Later we will add list of quotes for each interview
+         if (($f = fopen($sourceFilePath, "r")) !== FALSE) {
+           $columnOrder = []; // used to track column locations since we are going entirely by name instead of order
+           $givenColumnNames = fgetcsv($f);
+           foreach ($givenColumnNames as $column) {
+               $formattedColumn = $this->cleanString($column);
+               if(in_array($formattedColumn,$formattedSourceNames)) {
+                   array_push($columnOrder,$formattedColumn);
+               } else {
+                   array_push($columnOrder, "unrecognized");
+                   array_push($columnNameWarnings["unrecognized"],["item" => [$column]]); // NOTE: use this format with "item" to make it easier to call card.xslt (unless you want to change that)
+               }
+           }
+
+           if(!in_array("attribution",$columnOrder)) array_push($columnNameErrors,["item" => [self::ERR_NO_ATTRIBUTIONS]]);
+           if(!in_array("identifier",$columnOrder)) array_push($columnNameErrors,["item" => [self::ERR_NO_IDENTIFIER]]);
+
+           for($i = 0; $i < count($formattedSourceNames); $i++) {
+               $formattedSourceName = $formattedSourceNames[$i];
+               $unformattedSourceName = self::BATCH_SOURCE_DATA[$i];
+               if (!in_array($formattedSourceName,$columnOrder))
+                 array_push($columnNameWarnings["expected"],["item" => [$unformattedSourceName]]);
+           }
+
+           // These are both major errors. There is no need to parse the rest of the sheet as uploading will not be allowed if one of these errors occurs
```

Comment:
Is this comment still at the right spot?

Reply:
@Sammidysam Good catch. You are right.

Reply:
Fixed